Documentation update (#482)
kendraio/kendraio-app: docs/workflow/blocks/form.rst

Display a form for data entry or editing.

Forms are displayed based on the UI Schema and JSON Schema provided. These can be provided inline within the block configuration, or loaded from an adapter.

> Kendrai Form is build on top of Angular Formly https://formly.dev/. Most of the form property and functionality can be looked up in Formly.

Author: To be fixed: Kendraio

Reviewer: ```suggestion
Kendraio Form is built on top of Angular Formly https://formly.dev/. Much of the form property and functionality can be looked up from Formly documentation.
```

        }
    },
    "uiSchema": {}
  }

Dynamic data and field titles
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To generate fields form given data, the data must be in a format that the form block can read.

Reviewer 2: Presumably this should be '**from** given data'?

Author: done

        ]
    }
  }

  // Save these data to a context, under the name `civilStatusData`
  {
      "type": "context-save",
      "key": "civilStatusData"
  }

  // Pass the selected option
  {
    "status": `2`
  }

  // Form config
  {
    "type": "form",
    "hasSubmit": "true", 
    "label": "Click to submit",
    "jsonSchema": {
        "type": "object",
        "properties": {
            "name": {
                "type": "string",
                "title": "Name",
                "default": ""
            },
            "status": {
              "title": "Civil Status",
              "$ref": "#/definitions/context/civilStatusData"
        }
    },
    "uiSchema": {}
  }

In the above example, as soon as the Flow is open, it should show a select input with `Divorce` already selected, and the option to change it to `Single`.

Special UI: Card to add and remove items dynamically
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
`type: array` display a special UI which allow to add/delete dynamically
the specified field with the specified properties. 

In the below example, the form will allow the user to create a group of 
two fields: A number and a string. 

> Clicking on the add button the user will be able to add more of these group of items,

Author: I rephrase the whole bit. Hopefully is clearer now

> or remove the already created one.

Author: see above comment

            "user_name": {
                "type": "string",
                "title": "name",
                "default": "The value of this field comes from dynamic data"
            },
            "user_surname": {
                "type": "string",
                "title": "surname",
                "default": "The value of this field comes from dynamic data"
            }
        }
    },
    "uiSchema": {}
  }

Select Input
^^^^^^^^^^^^

> Display a field as select with several options.

Reviewer: ```suggestion
A field to allow the user to `select` input from a list of options.
```

Nested flows
------------

> If you want fixedto update a field value according to a user action, you can achieve this with a nested flow.

Reviewer: This typo still exists @gsambrotta
